domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java

```diff
@@ -18,29 +18,41 @@
     private static final String NO_STATEMENTS_MADE = "No statements have yet been made during that settlement";
     private final List<PartyStatement> partyStatements = new ArrayList<>();
 
-    public void makeOffer(Offer offer, MadeBy party) {
+    public void makeOffer(Offer offer, MadeBy party, String partyStatementId) {
```

Comment:
Suggestion - It may be a bad suggestion to have Optional in the method signature. But it will make code look neat. instead of sending null in lot of places, we can send Optional.empty.

Reply:
I thought of it, we don't pass Optional in method signature, but handle null using Optional in method